scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala

```diff
@@ -31,8 +31,8 @@ trait GrantHandler {
       }.getOrElse {
         handler.createAccessToken(authInfo)
       }
-      case Some(token) => Future.successful(token)
-      case None => handler.createAccessToken(authInfo)
+      case Some(token) if !token.isExpired => Future.successful(token)
```

Comment:
A developer is using Implicit grant, which doesn't support refresh token. Then he always get access token even if the token is expired. He doesn't want to return the expired token to users. So, I try to fix the problem.

Shouldn't we support re-issue access token automatically?
User can check an expired token on `getStoredAccessToken` by using `token.isExpired`. 

@rmmeans What do you think?